packages/oidc/src/token.ts

```diff
@@ -0,0 +1,226 @@
+interface TokenPayload {
+  sub: string;
+  name: string;
+  exp: number;
+}
+
+function isNodeEnv(): boolean {
+  return (
+    typeof process !== 'undefined' &&
+    process.versions != null &&
+    process.versions.node != null
+  );
+}
+
+function isBrowserEnv(): boolean {
+  return typeof window !== 'undefined';
+}
+
+async function getVercelDataDir(): Promise<string | null> {
+  if (isBrowserEnv()) {
+    return '';
+  }
+  const vercelFolder = 'com.vercel.cli';
+  const os = await import('os');
+  const path = await import('path');
```

Comment:
Parallelize these?
```suggestion
  const [os, path] = await Promise.all([
    import('os'),
    import('path'),
  ]);
```

Also, we have to do this because the build breaks if we put them statically at the top-level? There are no browser shims that get inserted for the browser case?

Could we do a synchronous `require` instead?

Reply:
I thought it was just browser but forgot that edge functions also don't have access to these as far as I understand